cmd/ddev/cmd/snapshot_test.go

```diff
@@ -1,46 +1,49 @@
 package cmd
 
 import (
-	"fmt"
 	"github.com/drud/ddev/pkg/ddevapp"
 	"github.com/drud/ddev/pkg/exec"
 	asrt "github.com/stretchr/testify/assert"
-	"os"
 	"testing"
 )
 
 // TestCmdSnapshot runs `ddev snapshot` on the test apps
 func TestCmdSnapshot(t *testing.T) {
 	assert := asrt.New(t)
 
-	testDir, _ := os.Getwd()
-	fmt.Println(testDir)
 	site := TestSites[0]
 	cleanup := site.Chdir()
+
 	app, err := ddevapp.NewApp(site.Dir, false, "")
 	assert.NoError(err)
+
 	defer func() {
 		// Make sure all databases are back to default empty
 		_ = app.Stop(true, false)
 		_ = app.Start()
 		cleanup()
 	}()
 
+	// Ensure that there are no snapshots available before we create one
+	args := []string{"snapshot", "--cleanup", "--yes"}
+	_, err = exec.RunCommand(DdevBin, args)
+	assert.NoError(err)
+
 	// Ensure that a snapshot can be created
-	args := []string{"snapshot", "--name", "test-snapshot"}
+	args = []string{"snapshot", "--name", "test-snapshot"}
 	out, err := exec.RunCommand(DdevBin, args)
 	assert.NoError(err)
-	assert.Contains(string(out), "Created snapshot test-snapshot")
+	assert.Contains(out, "Created snapshot test-snapshot")
 
 	// Try to delete a not existing snapshot
 	args = []string{"snapshot", "--name", "not-existing-snapshot", "--cleanup", "--yes"}
 	out, err = exec.RunCommand(DdevBin, args)
 	assert.Error(err)
-	assert.Contains(string(out), "Failed to delete snapshot")
+	assert.Contains(out, "Failed to delete snapshot")
 
 	// Ensure that an existing snapshot can be deleted
-	args = []string{"snapshot", "--name", "test-snapshot", "--cleanup"}
+	args = []string{"snapshot", "--name", "test-snapshot", "--cleanup", "--yes"}
 	out, err = exec.RunCommand(DdevBin, args)
 	assert.NoError(err)
-	assert.Contains(string(out), "Deleted database snapshot test-snapshot")
+	assert.Contains(out, "Deleted database snapshot test-snapshot")
```

Comment:
Let's add a test that tests the (fantastic) new prompted restore. It's not too hard to do

`echo "^[[B^M" | ddev snapshot restore`

Works at least on macOS. If it doesn't work on Windows, we can skip it on windows, although it would be nice to have coverage. Wonderful interactive things like this tends to rot and then nobody knows about it.

Reply:
I think it's not to easy, because `echo` is not callable by `exec.RunCommand`. It's part of the shell.

Reply:
In TestCmd tests, you can exec `bash -c "echo ... | ddev snapshot restore"`

Reply:
That's working on linux.
```go
exec.RunCommand("bash", []string{"-c", "echo -nq '\n' | " + DdevBin + " snapshot restore"})
```

Let's see if that also works on Mac.
I wrapped it in a `if runtime.GOOS != "windows" { ...`